src/client/App.tsx

```diff
@@ -1,5 +1,199 @@
+import { ChangeEvent, useState, useRef } from "react";
+
+import {
+  useGetPatients,
+  useDeletePatient,
+  usePostPatient,
+  usePutPatient,
+} from "./hooks";
+import { Patient, Severity, Status } from "./lib/type";
+
 const App = () => {
-  return <></>;
+  const [data, setData] = useState<string>("");
+  const [patient, setPatient] = useState<Patient>({} as Patient);
+  const patients = useGetPatients();
+  const createPatient = usePostPatient();
+  const updatePatient = usePutPatient();
+  const deletePatient = useDeletePatient();
+  const ref = useRef<HTMLDialogElement>(null);
+
+  const onFileChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files?.length) {
+      const reader = new FileReader();
+      reader.onload = async (e) => {
+        const text = e.target?.result;
+        setData(text as string);
+      };
+      reader.readAsText(e.target.files[0]);
+    } else {
+      setData("");
+    }
+  };
+  const onUpload = async () => {
+    if (!data) {
+      alert("No file to upload");
+      return;
+    }
+    const patientData = JSON.parse(data);
+    await createPatient.mutateAsync({
+      id: "",
+      patientId: patientData.PatientID,
+      name: patientData.PatientName,
+      dob: patientData.PatientDOB,
+      acquisitionDate: patientData.AcquisitionDate,
+      evaluationResult: patientData.EvaluationResult,
+      severity: patientData.Severity,
+      status: patientData.StatusReview,
+      comments: patientData.Comments,
+    });
+  };
+
+  const onClickRow = (data: Patient) => {
+    setPatient(data);
+    ref.current?.showModal();
+  };
+  const onChangeName = (e: ChangeEvent<HTMLInputElement>) => {
+    setPatient({ ...patient, name: e.target.value });
+  };
+  const onChangeDob = (e: ChangeEvent<HTMLInputElement>) => {
+    setPatient({ ...patient, dob: e.target.value });
+  };
+  const onChangeAcquisitionDate = (e: ChangeEvent<HTMLInputElement>) => {
+    setPatient({ ...patient, acquisitionDate: e.target.value });
+  };
+  const onChangeEvaluationResult = (e: ChangeEvent<HTMLInputElement>) => {
+    setPatient({ ...patient, evaluationResult: e.target.value });
+  };
+  const onChangeSeverity = (e: ChangeEvent<HTMLSelectElement>) => {
+    setPatient({ ...patient, severity: e.target.value as Severity });
+  };
+  const onChangeStatusReview = (e: ChangeEvent<HTMLSelectElement>) => {
+    setPatient({ ...patient, status: e.target.value as Status });
+  };
+  const onChangeComments = (e: ChangeEvent<HTMLInputElement>) => {
+    setPatient({ ...patient, comments: e.target.value });
+  };
+  const onUpdate = async () => {
+    await updatePatient.mutateAsync(patient);
+    ref.current?.close();
+  };
+  const onDelete = async (id: string) => {
+    await deletePatient.mutateAsync(id);
```

Comment:
Use parse mutateAsync from the mutation hook through object destructuring since the original instance tends change frequently.